### PR TITLE
[BugFix] Fix NPE when export job waiting coordinator profile finish

### LINKED ISSUE #14709: [BUG] Export job throws NPE when waiting coordinator profile finish
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
2. INSERT INTO '....'
3. SELECT  '....'

### Real behavior (Required)
```
2022-12-06 16:20:16,827 ERROR (export_exporting_sub_task_pool-0|4836) [MasterTask.run():33] task exec error
java.lang.NullPointerException: null
        at com.starrocks.qe.Coordinator.endProfile(Coordinator.java:1698) ~[starrocks-fe.jar:?]
        at com.starrocks.task.ExportExportingTask$ExportExportingSubTask.exec(ExportExportingTask.java:322) ~[starrocks-fe.jar:?]
        at com.starrocks.task.MasterTask.run(MasterTask.java:31) [starrocks-fe.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_322]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_322]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_322]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_322]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_322]
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by gengjun-git**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #14709

## Problem Summary(Required) ：

connectContext can be null for broker export task coordinator.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 3 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|0|3|00.00%|[1453, 1454, 1456]|

File: fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java [hunk 1]
```diff
@@ -1448,7 +1448,13 @@ public void endProfile() {
         // wait for all backends
         if (needReport) {
             try {
-                int timeout = connectContext.getSessionVariable().getProfileTimeout();
+                int timeout;
+                // connectContext can be null for broker export task coordinator
+                if (connectContext != null) {
+                    timeout = connectContext.getSessionVariable().getProfileTimeout();
+                } else {
+                    timeout = 2;
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java:1456]: what's the meaning of 2? shall we use a constant?

**Review by nshangyiming — DISMISSED** [review]

**Review by liuyehcf — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|2|50.00%|[1458]|

**Review by Astralidea — DISMISSED** [review]

**Review by nshangyiming — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by murphyatwork — APPROVED** [review]

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 3 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|0|3|00.00%|[1445, 1446, 1448]|

**Comment by gengjun-git** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Comment by gengjun-git** [comment]
@mergify backport branch-2.5 branch-2.4 branch-2.3

**Pull request merged**